### PR TITLE
added latest changes to cover bug 32913

### PR DESCRIPTION
**Pull request by Chronicless**

### CODE REVIEW

File: src/api/decode.js [hunk 1]
```diff
@@ -1,244 +1,340 @@
...
- * @author: Dmitry Kudryavtsev
- */
-var event_type_dict ={
-    0:'SYSTEM',
-    1:'UI',
-    2:'SIP',
-    3:'RUNTIME',
-    4:'HM',
-    10:'UNDEFINED'
+var eventTypeDict = {
```
Comment (rfmines) [inline src/api/decode.js:1]: This file needs to go away and we need an initialization script that loads the values into mongo if they dont exist. Please remove decode.js from the repo, create the initialization script and update the PR

File: src/test/supportMethods/encodeBody.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+var moment = require('moment');
+
+/* exports.encodeBody =*/
+// encoding body in the same way as mobile app do this
+// note : encodeBody invokes itself (recursion)
+function encodeBody(body) {
+    var decodedBody = {};
+    for (var iter in body) {
+        // looking for field short name , for HTTP req
+        var decodedKey = htmlSymbols[iter];
```
Comment (rfmines) [inline src/test/supportMethods/encodeBody.js:14]: what is the purpose of this?

File: src/api/dictionary.js [hunk 3]
```diff
@@ -0,0 +1,105 @@
+'use strict';
+var htmlBodyValues = {
```
Comment (rfmines) [inline src/api/dictionary.js:2]: what is the purpose of this file ?

File: src/api/decode.js (hunk 1 above)
Comment (Chronicless) [reply src/api/decode.js:1]: okay , but in this case we have to create also getters for this from mongo and change api methods implementation , is it ok?

File: src/test/supportMethods/encodeBody.js (hunk 2 above)
Comment (Chronicless) [reply src/test/supportMethods/encodeBody.js:14]: this is for testing current version of data upload from mobile side , it is encodes http body with events in the same way with the same rules as mobile app do that before send data to csl

File: src/api/dictionary.js (hunk 3 above)
Comment (Chronicless) [reply src/api/dictionary.js:2]: this is dictionary with current body encoding rules(described in csl hld) , which mobile app uses right now and list of possible events with array of possible lables (labelids)(from csl hld too).

File: src/api/decode.js (hunk 1 above)
Comment (rfmines) [reply src/api/decode.js:1]: there is an api to get and set these in the master branch on git. After that i think its just a mongo lookup?

File: src/api/dictionary.js (hunk 3 above)
Comment (rfmines) [reply src/api/dictionary.js:2]: this file doesnt exist in the current branch. why are you adding it?

Comment (Chronicless) [reply src/api/dictionary.js:2]: first of all this is using in test case , when we generate an array of events this array must contain correct labels for each event_type and name(second dictionary), then we have to encode it  before sending to CSL (first dict).
Second purpose , this is will be a part of dictionary with CSL measurements,requested by Afshin (dictionary in mongo with all events , its possible lables etc,)
We can split this files into 2 - first move to test(encoding values) and the second dictionary (with current lables for each event ) we can move in mongo as a part of dictionary with measurements .

**Comment by Chronicless** [comment]
some node modules was deleted with old version and new one was not uploaded , i will fix it

**Pull request merged**